stock_available_to_promise_release/models/stock_move.py

```diff
@@ -622,6 +670,30 @@ def unrelease(self, safe_unrelease=False):
                     origin_moves.write({"propagate_cancel": False})
                     # origin_moves._action_cancel()
                     moves_to_cancel |= origin_moves
+                # checking that for the current step (pick/pack/ship)
+                # move.product_uom_qty == step.cancelled_qty + move.returned_quanty
+                # If not the case, we have to move back goods in stock.
+                qty_cancelled_at_step = sum(origin_moves.mapped("product_uom_qty"))
```

Comment:
Compute in the uom of the product as each move could have a different uom
```suggestion
                qty_cancelled_at_step = sum(origin_moves.mapped("product_qty"))
```